README.md

```diff
@@ -103,57 +103,126 @@ FLP is a framework. The main FLP object is the **pipeline**. FLP **pipeline** ca
 
 # Architecture
 
-The pipeline is constructed of a sequence of stages:
+The pipeline is constructed of a sequence of stages. Each stage is classified into one of the following classes:
 - **ingest** - obtain flows from some source, one entry per line
 - **decode** - parse input lines into a known format, e.g., dictionary (map) of AWS or goflow data
 - **transform** - convert entries into a standard format; can include multiple transform stages
 - **write** - provide the means to write the data to some target, e.g. loki, standard output, etc
 - **extract** - derive a set of metrics from the imported flows
 - **encode** - make the data available in appropriate format (e.g. prometheus)
 
-The **encode** and **write** stages may be combined in some cases (as in prometheus), in which case **write** is set to **none**
+The first stage in a pipeline must be an **ingest** stage, followed by a **decode** stage.
```

Comment:
Can we have more than one decode ... it is not clear from the text?

Reply:
Is the updated text now better?

Reply:
I think so :-)